client/src/components/ProductDetails/ImageGallery.jsx

```diff
@@ -1,30 +1,46 @@
-import React, { useState } from "react";
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react';
 import './prodStyles.css';
 
-function ImageGallery ({imageGallery}) {
+function ImageGallery ({styles, currentProductID, currentStyleID}) {
 
-  const [currentImage, updateMainImage] = useState(imageGallery[0])
+  const getCurrentProductObject = (targetId, allProducts) => {
+    return allProducts.filter((product) => {
+      return product.product_id === targetId;
+    });
+  };
 
-  const setThumbnailClass = (image) => {
-    return image === currentImage ? "imageThumbnailMain" : "imageThumbnail"
-  }
-  const enterExpandedView = () => {
+  const currentProductObject = getCurrentProductObject(currentProductID, styles);
+
+  const getStylesArray = (targetStyleId, stylesArray) => {
+    return stylesArray.filter((style) => {
+      return style.style_id === targetStyleId;
+    });
+  };
+
+  const stylesArray = currentProductObject[0]['results'];
```

Comment:
What happens if currentProductObject[0] is undefined

Reply:
I have not decided on how to handle this edge case. This would only happen if there is an error with the GET req to the API. By default the first product Object returned is to be set to currentProductObject